src/cli.ts

```diff
@@ -59,10 +62,93 @@ const noop = (): void => {
     logger.info('noop performed');
 };
 
+
+
+const syncBadPathCounts = async (): Promise<void> => {
+    await useDatabaseTransaction(async () => {
+        const topics = await CourseTopicContent.findAll(
+            {
+                where: {
+                    active: true,
+                },
+                include: [
+                    {
+                        model: CourseWWTopicQuestion,
+                        as: 'questions',
+                        include: [
+                            {
+                                model: CourseQuestionAssessmentInfo,
+                                as: 'courseQuestionAssessmentInfo'
+                            }
+                        ]
+                    }
+                ]
+            }
+        );
+
+        logger.info(`Checking ${topics.length} topics`);
+
+        await topics.asyncForEach(async (topic) => {
+            // For each question in the topic, sum the existence of a path problem.
+            let count = 0;
+            await topic.questions?.asyncForEach(async (question)=>{
+                const {webworkQuestionPath, errors, courseQuestionAssessmentInfo} = question;
+                const additionalProblemPaths = courseQuestionAssessmentInfo?.additionalProblemPaths;
+                const additionalErrors = courseQuestionAssessmentInfo?.errors;
+
+                const isDefinitelyBad = VALID_PG_PATH_REGEX.test(webworkQuestionPath) === false;
+
+                // If this is definitely bad and not already in the error object, add it.
+                if (isDefinitelyBad && (_.isNil(errors) || _.isEmpty(errors) || !(webworkQuestionPath in errors))){
+                    logger.debug(`Updating ${question.id} from topic ${topic.id} with a path error.`);
+                    question.errors = {
```

Comment:
If there was already errors wouldn't this override it
Shouldn't question.errors be spread into the object in case it is not null?

Reply:
Oh, this is a good catch. Right now, it doesn't matter because path errors are the only errors we use this field for, but you're right.